src/TCommandLine.cpp

```diff
@@ -672,192 +672,202 @@ void TCommandLine::slot_popupMenu()
+                                            "used when the command spelling checker using the dictionary shared between profile has no words to suggest"));
             } else {
-                popup->insertActions(separator_aboveStandardMenu, spellings_system);
+                pA = new QAction(tr("no suggestions (profile)",
+                                            // Intentional comment
+                                            "used when the command spelling checker using the profile's own dictionary has no words to suggest"));
             }
-            // else the word is in the dictionary - in either ca`se show the context
-            // menu - either the one with the prefixed spellings, or the standard
-            // one:
+            pA->setEnabled(false);
+            spellings_profile << pA;
+        }
+    }
+
+    /*
+    * Build up the extra context menu items from the BOTTOM up, so that
+    * the top of the context menu looks like:
+    *
+    * profile dictionary suggestions
+    * --------- separator_aboveDictionarySeparatorLine
+    * \/ System dictionary suggestions /\ Profile  <== Text
+    * --------- separator_aboveSystemDictionarySuggestions
+    * system dictionary suggestions
+    * --------- separator_aboveAddAndRemove
+    * Add word action
+    * Remove word action
+    * --------- separator_aboveStandardMenu
+    *
+    * The insertAction[s](...)/(Separator(...)) insert their things
+    * second argument (or generated by themself) before the first (or
+    * only) argument given.
+    */
+
+    auto separator_aboveStandardMenu = popup->insertSeparator(popup->actions().first());
+    if (handle_profile) {
+        popup->insertAction(separator_aboveStandardMenu, action_removeWord);
+        popup->insertAction(action_removeWord, action_addWord);
+        auto separator_aboveAddAndRemove = popup->insertSeparator(action_addWord);
+        popup->insertActions(separator_aboveAddAndRemove, spellings_system);
+        auto separator_aboveSystemDictionarySuggestions = popup->insertSeparator(spellings_system.first());
+        popup->insertAction(separator_aboveSystemDictionarySuggestions, action_dictionarySeparatorLine);
+        auto separator_aboveDictionarySeparatorLine = popup->insertSeparator(action_dictionarySeparatorLine);
+        popup->insertActions(separator_aboveDictionarySeparatorLine, spellings_profile);
+    } else {
+        popup->insertActions(separator_aboveStandardMenu, spellings_system);
+    }
+}
+
+void TCommandLine::mousePressEvent(QMouseEvent* event)
+{
+    if (event->button() == Qt::RightButton) {
+#if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
+        auto popup = createStandardContextMenu(event->globalPos());
+#else
+        auto popup = createStandardContextMenu(event->globalPosition().toPoint());
+#endif
+        if (mpHost->mEnableSpellCheck) {
+            fillSpellCheckList(event, popup);
+            // else the word is in the dictionary - in either case show the context
```

Comment:
:confused:  This comment doesn't make sense in this position now...

Reply:
Would you suggest it makes more sense to move its position to the end of TCommandLine::fillSpellCheckList() then?